testcases/kernel/device-drivers/uart/serialcheck.sh

```diff
@@ -0,0 +1,95 @@
+#! /bin/sh
+###############################################################################
+# Copyright (C) 2015 Texas Instruments Incorporated - http://www.ti.com/
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation version 2.
+#
+# This program is distributed "as is" WITHOUT ANY WARRANTY of any
+# kind, whether express or implied; without even the implied warranty
+# of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+###############################################################################
+
+source "common.sh"  # Import do_cmd(), die() and other functions
+
+#### Functions definitions ####
+usage()
+{
+    echo "usage: ./${0##*/} [-r UART_RATE] [-l LOOPS] [-x to enable HW flow control]"
+    exit 1
+}
+
+check_requirements() {
+    which serialcheck || die "serialcheck binary is not available. Please install it"
+}
+
+create_test_file() {
+    temp_test_file=`mktemp`
+    dd if=/dev/urandom of=$temp_test_file count=1 bs=$((UART_RATE / 2))
+}
+
+get_uart_ports() {
+for i in ${ARRAY[@]}; do
+    PORT=/dev/`echo $i | cut -d'/' -f 5`
+    # Activate port in case it will be initialized only when startup
+    echo "DDT TESTING" > $PORT 2>/dev/null
```

Comment:
So this writes "DDT TESTING" to all ports found on the SUT?

Reply:
Only on the ones that are not being used, for example, it won't do it on your serial console port.